sway-core/src/asm_generation/register_allocator.rs

```diff
@@ -110,7 +110,7 @@ pub(crate) fn liveness_analysis(ops: &[RealizedOp]) -> HashMap<usize, BTreeSet<V
             }
 
             // Did anything change in this iteration?
-            modified = (prev_live_in_op != *live_in_op) || (prev_live_out_op != *live_out_op);
+            modified |= (prev_live_in_op != *live_in_op) || (prev_live_out_op != *live_out_op);
```

Comment:
I'm just noticing now what you actually did there -- a while loop with everything, all side-effects, in the condition.  I've never seen that before... it's a little confusing.

Why not just  `while !modified { .. }`?

Reply:
Apparently, that's how a "do while" look like in Rust.

I can make it clearer when I land.

Reply:
Hrrm.
```rust
let mut modified = false;
while !modified {
    ...
    modified = check...
}
```
looks more straight forward to me.

Also, while I'm here and nitpicking, all of the initialisation of `HashMap`s could be done with `FromIter`.
e.g.,
```rust
let offset_to_ix = HashMap::from_iter(ops.iter().enumerate().map(|(idx, op)| (op.offset, idx)));
``` 
This may be preferable because you don't need to declare `offset_to_ix` as mutable.

But also:
```rust
let mut live_in: HashMap<usize, BTreeSet<VirtualRegister>> =
     HashMap::from_iter((0..ops.len()).into_iter().map(|idx| (idx, BTreeSet::new())));
```
In this case it needs to be mutable anyway, so perhaps there's no point.

Reply:
(This is me embracing my functional programming sensibilities and attempting to shed decades of imperative-programming-in-C++ baggage...)

Reply:
Tell me about it! Not decades, but `C++` is all I've known 😆 

Anyways, I will address your comments above asap.. I created an issue here: https://github.com/FuelLabs/sway/issues/1055 and I'll get it towards the end of this week as I'm mostly focusing on the workshop now.